CRM/Contact/BAO/ContactType.php

```diff
@@ -603,7 +603,7 @@ public static function add(&$params) {
 
     if (!empty($params['id'])) {
       $newParams = [
-        'label' => "New $contact",
+        'label' => ts("New %1", [1 => $contact]),
```

Comment:
This line only handles editing an existing type. I think you also need it down on line 619.

Reply:
Indeed, thanks!